include/particle.h

```diff
@@ -262,6 +262,17 @@ class Particle : public ParticleBase<Tdim> {
   //! Assign material id of this particle to nodes
   void append_material_id_to_nodes() const override;
 
+  //! Return the number of neighbour particles
+  unsigned nneighbours() const override { return neighbours_.size(); };
```

Comment:
What is the intended use of `nneighbours`, could we not use `neighbours` and get the size?